src/lib/fields_derivers_graphql/fields_derivers_graphql.ml

```diff
@@ -0,0 +1,335 @@
+open Core_kernel
+open Fieldslib
+
+(*
+module Graphql_args = struct
+  module Input = struct
+    open Graphql_async
+
+    type 'input_type t = 'input_type Schema.Arg.arg_typ
+  end
+
+  module Creator = struct
+    open Graphql_async
+
+    type 'input_type t = String.Map.t
+
+    (*
+    type 'input_type t =
+      | E :
+          ('input_type, 'fields) Schema.Arg.arg_list * 'fields
+          -> 'input_type t
+          *)
+  end
+
+  module Output = struct
+    type 'input_type t = 'input_type option Input.t
+  end
+
+  module Accumulator = struct
+    type 'input_type t = unit
+  end
+
+  let init () = ()
+
+  let add_field (type f input_type) :
+         f Input.t
+      -> ( [< `Read | `Set_and_create ]
+         , input_type
+         , f )
+         Fieldslib.Field.t_with_perm
+      -> input_type Accumulator.t
+      -> (input_type Creator.t -> f) * input_type Accumulator.t =
+   fun _t_field _field _acc ->
+    ((fun (E (args, coerce)) -> failwith "Unused"), ())
+
+  let finish ((_creator, _x) : 'u * 'input_type Accumulator.t) :
+      'input_type Output.t =
+    failwith "TODO"
+end
+*)
+
+module Graphql_fields_raw = struct
+  module Make (IO : Graphql_intf.IO) = struct
+    module Schema = Graphql_schema.Make (IO)
+
+    module Deriver_basic = struct
+      module Input = struct
+        type 'input_type t =
+          { run : 'ctx. ?doc:string -> unit -> ('ctx, 'input_type) Schema.typ }
```

Comment:
We don't use the `?doc` argument for anything internally, should we just drop it here?

Reply:
we need it for composition since we use it on the finish bit -- we can sort this out tomorrow